contracts/pcv/maker/MakerRouter.sol

```diff
@@ -0,0 +1,229 @@
+    /// @notice Function to withdraw tokens to an address
+    /// @param token the token to withdraw
+    /// @param amount the amount to send
+    /// @param to the address the token should be sent to
+    function withdrawERC20(
+        address token,
+        uint256 amount,
+        address to
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        IERC20(token).safeTransfer(to, amount);
+    }
+
+    /// @notice Helper function to redeem DAI from the FEI FEI-DAI PSM
+    /// @param amountFeiIn the amount of FEI to be deposited
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param to the address the DAI should be sent to once swapped
+    function _redeemFromFeiPSM(
+        uint256 amountFeiIn,
+        uint256 minDaiAmountOut,
+        address to
+    ) private {
+        require(amountFeiIn > 1e18, "MakerRouter: Must deposit at least 1 FEI");
+        fei.safeTransferFrom(msg.sender, address(this), amountFeiIn);
+        feiPSM.redeem(to, amountFeiIn, minDaiAmountOut);
+    }
+
+    /// @notice Helper function to redeem DAI max allowance or balance from the FEI FEI-DAI PSM
+    /// @param to the address the DAI should be sent to once swapped
+    function _redeemAllBalanceFromFeiPSM(address to)
+        private
+        returns (uint256 minDaiAmountOut)
+    {
+        uint256 allowance = fei.allowance(msg.sender, address(this));
+        uint256 amountFeiIn = Math.min(fei.balanceOf(msg.sender), allowance);
+
+        require(amountFeiIn > 1e18, "MakerRouter: Must deposit at least 1 FEI");
+
+        fei.safeTransferFrom(msg.sender, address(this), amountFeiIn);
+        uint256 userBalanceBefore = dai.balanceOf(to);
+
+        // 3 is used as there is a 3 basis points redemption fee
+        minDaiAmountOut =
+            (amountFeiIn * (Constants.BASIS_POINTS_GRANULARITY - 3)) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        feiPSM.redeem(to, amountFeiIn, minDaiAmountOut);
+
+        uint256 userBalanceAfter = dai.balanceOf(to);
+
+        require(
+            userBalanceAfter - userBalanceBefore >= minDaiAmountOut,
+            "MakerRouter: Not enough DAI received"
+        );
+    }
+
+    /// @notice Helper function to swap for both DAI and USDC
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param usdcTo the address the USDC should be sent to once swapped
+    /// @param daiTo the address the DAI should be sent to once swapped
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+
+    function _swapForUsdcAndDai(
+        uint256 minDaiAmountOut,
+        address usdcTo,
+        address daiTo,
+        uint256 ratioUSDC
+    ) private {
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        require(
+            usdcAmount / USDC_SCALING_FACTOR != 0,
```

Comment:
Is it cheaper gas wise to compute `usdcAmount / USDC_SCALING_FACTOR` once and store the result on the stack as a local variable?

Reply:
I think one extra DIV operation is fine